File: BOJ 2/11055.cpp
#include <cstdio>
#include <algorithm>
using namespace std;
int n,Max;
int arr[1003];
int dp[1003];
int main(){
	scanf("%d", &n);
	for (int i = 1; i <= n; i++) {
		scanf("%d", &arr[i]);
		dp[i] = arr[i];
	}
	Max = dp[1];
	for (int i = 2; i <= n; i++){
		for (int j = 1; j < i; j++) {
			if (arr[j]<arr[i]) {
				dp[i] = max(dp[i], arr[i] + dp[j]);
			}
		}
		Max = max(dp[i], Max);
	}
	printf("%d\n", Max);
	return 0;
}

File: BOJ(Python 공부용)/11055.py
import sys

input = sys.stdin

if __name__ == '__main__':
    n = int(input.readline())
    arr = [int(x) for x in input.readline().split()]
    dp = [0] * (n + 1)
    ans = arr[0]
    for i in range(n):
        dp[i] = arr[i]
        for j in range(0, i):
            if arr[i] > arr[j]:
                dp[i] = max(dp[i], dp[j] + arr[i])
                ans = max(ans, dp[i])

    print(ans)
